pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -33,9 +33,17 @@ def on_trainer_init(
         check_val_every_n_epoch: int,
         reload_dataloaders_every_n_epochs: int,
         reload_dataloaders_every_epoch: bool,
-        prepare_data_per_node: bool,
+        prepare_data_per_node: Optional[bool] = None,
     ) -> None:
         self.trainer.datamodule = None
+
+        if prepare_data_per_node is None:
+            prepare_data_per_node = True
```

Comment:
we don't have to set this as True by default. we can leave it as optional, as this will allow the data hooks to be the primary source of truth